xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-templates/src/main/resources/templates/rating_macros.vm

```diff
@@ -36,7 +36,9 @@
     <div class="rating-header $cssclass">$ratinglabel</div>
     <div class="rating-stars">
         <ul class="star-rating ${cssclass}-star #if($locked) locked #end">
-            <li class="current-rating" style="width:${width}%;"></li>
+            <li class="current-rating">
+                <meter class="average-rating" min="0" max="5" value="$rating"/>
+            </li>
```

Comment:
Very interesting case of "is this a potentially breaking DOM change or not?" :) I mean: here the chance that anyone is impacted is absurdly low so I don't see any reason to not accept the changes. But, in theory we could imagine somehow relying on some code to check the width of a `current-rating` class in javascript to perform some stuff...

Reply:
So as @Sereza7 mentioned in the chat: 
> It's been a while that we have don't use inline style declarations in [our front end code style](https://dev.xwiki.org/xwiki/bin/view/Community/CodeStyle/XhtmlCssCodeStyle/#HCSS)

so then it's clearly not an API we're breaking :)

Reply:
(more discussion from the chat, moved here for the sake of keeping long term traces)

> I think this rating_macros.vm is also used to display ratings in pages, so it's not only for the admin
it's just that you don't have the Ratings Extension bundled by default so you need to install it first: you might want to check that

I just finished checking simple interactions, and it seems to behave like the one in extension behaved. I made minimal modifications, only removing the default inline styling and add a meter node in the list item. The part that would be difficult to handle for backward compatibility (handling how we update this at runtime) is not changed. It's not perfect, but the goal of this ticket is to avoid breaking web standard tests (only run at page load) and backward compatibility.